sentry/sentry.go

```diff
@@ -0,0 +1,214 @@
+	}
+}
+
+// TraceEnabled allows to change wether (the somewhat expensive, relatively)
+// stack trace generation takes place.
+func TraceEnabled(enabled bool) Option {
+	return func(l *Logger) {
+		l.traceEnabled = enabled
+	}
+}
+
+// TraceCfg allows to change the number of skipped frames, number of context lines and
+// list of go prefixes that are considered "in-app", i.e. "github.com/uber-go/zap".
+func TraceCfg(skip, context int, prefixes []string) Option {
+	return func(l *Logger) {
+		l.traceSkipFrames = skip
+		l.traceContextLines = context
+		l.traceAppPrefixes = prefixes
+	}
+}
+
+// Extra stores additional information for each Sentry event.
+func Extra(extra map[string]interface{}) Option {
+	return func(l *Logger) {
+		l.extra = extra
+	}
+}
+
+// Log sends Sentry information provided minimum threshold is met.
+func (l *Logger) Log(lvl zap.Level, msg string, fields ...zap.Field) {
+	l.log(lvl, msg, fields)
+}
+
+// Debug is a nop.
+func (l *Logger) Debug(msg string, fields ...zap.Field) {}
+
+// Info sends Sentry information provided minimum threshold is met.
+func (l *Logger) Info(msg string, fields ...zap.Field) {
+	l.log(zap.InfoLevel, msg, fields)
+}
+
+// Warn sends Sentry information provided minimum threshold is met.
+func (l *Logger) Warn(msg string, fields ...zap.Field) {
+	l.log(zap.WarnLevel, msg, fields)
+}
+
+// Error sends Sentry information provided minimum threshold is met.
+func (l *Logger) Error(msg string, fields ...zap.Field) {
+	l.log(zap.ErrorLevel, msg, fields)
+}
+
+// Panic sends Sentry information provided minimum threshold is met, then panics.
+func (l *Logger) Panic(msg string, fields ...zap.Field) {
+	l.log(zap.PanicLevel, msg, fields)
+	panic(msg)
+}
+
+// Fatal sends Sentry information provided minimum threshold is met, then exits.
+func (l *Logger) Fatal(msg string, fields ...zap.Field) {
+	l.log(zap.FatalLevel, msg, fields)
+	os.Exit(1)
+}
+
+// DFatal is a nop.
+func (l *Logger) DFatal(msg string, fields ...zap.Field) {}
+
+// With returns Sentry logger with additional context.
+func (l *Logger) With(fields ...zap.Field) zap.Logger {
+	// Consider pre-allocing the whole new resulting map, but that may be even more expensive
+	for _, f := range fields {
+		f.AddTo(l.extra)
+	}
+	return l
+}
+
+// Notify sentry if the log level meets minimum threshold.
+func (l *Logger) log(lvl zap.Level, msg string, fields []zap.Field) {
+	if lvl >= l.minLevel {
+		// append all the fields from the current log message to the stored ones
+		extra := l.extra
+		for _, f := range fields {
+			f.AddTo(extra)
+		}
+
+		packet := &raven.Packet{
+			Message:   msg,
+			Timestamp: raven.Timestamp(time.Now()),
+			Level:     _zapToRavenMap[lvl],
+			Platform:  _platform,
+			Extra:     extra,
+		}
+
+		if l.traceEnabled {
+			trace := raven.NewStacktrace(l.traceSkipFrames, l.traceContextLines, l.traceAppPrefixes)
+			if trace != nil {
+				packet.Interfaces = append(packet.Interfaces, trace)
+			}
+		}
+
+		l.Capture(packet)
```

Comment:
Fine for now. You may want to talk to the upstream `raven-go` maintainers to see if they're willing to expose a more performant API. Perhaps `CaptureBytes([]byte)`, so we can use zap's JSONEncoder and incrementally serialize this payload, passing the JSON-encoded bytes here?

/cc @sectioneight